app/validators/questions/question_validator.py

```diff
@@ -23,15 +23,11 @@ def _validate_answer_labels(self):
         if len(self.answers) < 2:
             return None
 
+        final_answer_single_checkbox = self.answers[1]["type"] == "Checkbox"
         for answer in self.answers:
-            if not answer.get("label"):
-                if (
-                    answer["type"] == "Checkbox"
-                    and (answer == self.answers[-1])
-                    and len(answer["options"]) < 2
-                ):
-                    return None
-
+            if not (
+                len(self.answers) == 2 and final_answer_single_checkbox
+            ) and not answer.get("label"):
```

Comment:
The checkbox is checked in every loop, even though it won't change. It should be done in a conditional outside of the loop. It should also check that it's a single checkbox:
```python
        if (
            len(self.answers) == 2
            and self.answers[-1]["type"] == "Checkbox"
            and len(self.answers[-1]["options"]) == 1
        ):
            return None
```